.github/actions/setup-monorepo/action.yml

```diff
@@ -0,0 +1,54 @@
+name: Setup Monorepo
+description: Prepare and install everything for the monorepo
+
+inputs:
+  node-version:
+    description: Version of Node to use
+    default: 18.x
+
+  pnpm-version:
+    description: Version of pnpm to use
+    default: 8.x
+
+  eas-version:
+    description: Version of EAS CLI to use
+    default: latest
+
+  expo-token:
+    description: Expo token to authenticate with
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: 🏗 Setup pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: ${{ inputs.pnpm-version }}
+
+    - name: 🏗 Setup Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node-version }}
+        cache: pnpm
+
+    - name: 🏗 Setup Expo
+      uses: expo/expo-github-action@v8
+      with:
+        eas-version: ${{ inputs.eas-version }}
+        token: ${{ inputs.expo-token }}
+
+    - name: 📦 Install dependencies
+      run: pnpm install
+      shell: bash
+
+    - name: ♻️ Restore cache
```

Comment:
Noob question: is it normal that we restore the cache *after* the other steps? Also, should we use `cache@v4`?

Reply:
The cache we are talking about is the Turbo cache (see the paths that are being restored below. So it doesn't matter where it is. I consulted this: https://github.com/byCedric/expo-monorepo-example/blob/main/.github/actions/setup-monorepo/action.yml

v4 uses node 20, I would say we go for v3 now and update our whole stack slowly to v20 (we still use v16 in some cases: https://github.com/alephium/alephium-frontend/issues/343

Reply:
Alrighty then! Let's try it live for test builds :)